src/data/selector.ts

```diff
@@ -822,3 +822,34 @@ export const handleLegacyDeviceSelector = (
     device: rest,
   };
 };
+
+export const computeCreateDomains = (
+  selector: EntitySelector | TargetSelector
+): undefined | string[] => {
+  let entityFilters: EntitySelectorFilter[] | undefined;
+
+  if ("target" in selector) {
+    entityFilters = ensureArray(selector.target?.entity);
+  } else if ("entity" in selector) {
+    if (selector.entity?.include_entities) {
+      return undefined;
+    }
+    entityFilters = ensureArray(selector.entity?.filter);
```

Comment:
Should we also take [LegacyEntitySelector](https://github.com/home-assistant/frontend/blob/b2797ab8da3c349ee19e6ec4a620ddb78d919345/src/data/selector.ts#L221) into account?

Reply:
I think it's already handled by the automatic conversion?

Gauge card uses legacy format in ha-form and I observed that it worked, and I tried a BP version of legacy format that seems to already work as well. 

```
    legacy_entity:
      selector:
        entity:
          domain:
            - counter
            - input_number
```